pkg/printer/checkstyle.go

```diff
@@ -0,0 +1,60 @@
+package printer
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/get-woke/woke/pkg/result"
+	"github.com/get-woke/woke/pkg/rule"
+)
+
+// Checkstyle is a Checkstyle printer meant for use by a Checkstyle annotation
+type Checkstyle struct {
+	writer io.Writer
+}
+
+// NewGitHubActions returns a new GitHubActions printer
+func NewCheckstyle(w io.Writer) *Checkstyle {
+	return &Checkstyle{writer: w}
+}
+
+func (p *Checkstyle) PrintSuccessExitMessage() bool {
+	return true
+}
+
+// Print prints in the format for Checkstyle
+// https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-error-message
+func (p *Checkstyle) Print(fs *result.FileResults) error {
+	fmt.Fprintf(p.writer, "  <file name=\"%s\">\n", fs.Filename)
+	for _, r := range fs.Results {
+		fmt.Fprintln(p.writer, formatResultForCheckstyle(r))
+	}
+	fmt.Fprintln(p.writer, `  </file>`)
+	return nil
+}
+
+func (p *Checkstyle) Start() {
+	fmt.Fprintln(p.writer, `<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
The xml package gives this first line to you with `xml.Header` - https://pkg.go.dev/encoding/xml#pkg-constants